### PR TITLE
🌱 WIP: Reformat logs/bootstrap 

### PR DESCRIPTION
**Pull request by killianmuldoon**
This PR is a work in progress and leads on from #5571. This is a poc to see how we should improve our logging overall and should not be merged.

Add extra key-value pairs and add logging levels to logs in the bootstrap package.

### CODE REVIEW

File: bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go [hunk 1]
```diff
@@ -508,6 +519,7 @@ func (r *KubeadmConfigReconciler) joinWorker(ctx context.Context, scope *Scope)
 	if res, err := r.reconcileDiscovery(ctx, scope.Cluster, scope.Config, certificates); err != nil {
 		return ctrl.Result{}, err
 	} else if !res.IsZero() {
+		log.Info("Reconcile finished - discovery configuration set")
```
Comment (fabriziopandini) [inline bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:522]: I will drop this in favor of logging inside reconcileDiscovery when we return non zero values

**Review by fabriziopandini — COMMENTED** [review]
Some nits, but i really like the direction of this PR

**Comment by killianmuldoon** [comment]
>@killianmuldoon do you still want to pursue these changes, or shall we close?
I think I'll come back to this over the next few weeks.

**Comment by killianmuldoon** [comment]
Closing this for now as the changes are stale / behind where we are on logging right now. Will possibly come back to this on a new PR, or open a new PR with similar changes.

/close

**Pull request closed without merging**